validate-config/validate-config.yaml

```diff
@@ -0,0 +1,67 @@
+name: Hub Config Validation (R)
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: main
+    paths:
+      - 'hub-config/**'
+      - '!**README**'
+
+jobs:
+  validate-hub-config:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          install-r: false
+          use-public-rspm: true
+
+      - name: Update R
+        run: |
+          sudo apt-get update
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          packages: hubverse-org/hubAdmin, any::sessioninfo
+
+      - name: Run validations
+        id: validate
+        env:
+          PR_NUMBER: ${{ github.event.number }}
+        run: |
+          workspace <- "${{ github.workspace }}"
+          output <- Sys.getenv("GITHUB_OUTPUT")
+          diff <- file.path(workspace, "diff.md")
+          v <- hubAdmin::validate_hub_config(
+            hub_path = workspace
+          )
+          # check if there were any failures
+          invalid <- any(vapply(v, isFALSE, logical(1)))
+          if (invalid) {
+            cat("result=false", "\n", file = output, sep = "", append = TRUE)
+            # write output to HTML
+            tbl <- hubAdmin::view_config_val_errors(v)
+            writeLines("## :x: Invalid Configuration\n", diff)
+            cat("\nErrors were detected in one or more config files in `hub-config/`. Details about the exact locations of the errors can be found in the table below.\n", file = diff, append = TRUE)
+            cat(gt::as_raw_html(tbl), "\n", file = diff, sep = "", append = TRUE)
+          } else {
+            cat("result=true", "\n", file = output, sep = "", append = TRUE)
+          }
+        shell: Rscript {0}
+      - name: "Comment on PR"
+        id: comment-diff
+        if: ${{ github.event_name != 'workflow_dispatch' && steps.validate.outputs.result == 'false' }}
+        uses: carpentries/actions/comment-diff@main
+        with:
+          pr: ${{ github.event.number }}
+          path: ${{ github.workspace }}/diff.md
```

Comment:
Should this also change if I want to set a custom hub_path?

Reply:
Yes. I can definitely set this up so that the hub path is set upstream in the job.